### PR TITLE
Updates course run and program certificate models to limit choices just to certificate pages in admin

### LINKED ISSUE #1283: Certificate revision page dropdown should display only certificate page revisions in Django admin
### Steps to Reproduce

- Create a certificate manually from Django admin for program or course run
- Click on the certificate page revision dropdown
- Here you’ll see page revisions of all pages

### Expected Behavior

- We need certificate page revision only in the dropdown

### Actual Behavior

- Getting page revisions for all pages in CMS

### Related Issues

#887

### PR DESCRIPTION
**Pull request by jkachel**
#### Pre-Flight checklist

- [X] Migrations
  - [X] Migration is backwards-compatible with current production code
- [X] Testing
  - [X] Code is tested
  - [X] Changes have been manually tested

#### What are the relevant tickets?

Closes #1283 

#### What's this PR do?

Updates the program and course run certificate models' certificate page foreign keys to limit their choices to actual certificate pages, rather than all CMS pages.

#### How should this be manually tested?

1. Ensure you have certificate pages set up properly.
2. Apply the migrations.
3. In Django Admin, edit or create either a program or a course run certificate. The options in the list for Certificate page revision should only be certificate pages. (Prior to this, all CMS pages were listed; you could set the cert page to be the MITx Online homepage, for instance.) 
4. Repeat step 3 with the courseware object type you haven't yet tested.

### CODE REVIEW

File: courses/models.py [hunk 1]
```diff
@@ -658,6 +659,23 @@ def save(
...
+    CMS.
+    """
+    content_type = ContentType.objects.filter(
+        app_label="cms", model="certificatepage"
+    ).get()
+
+    avaialble_revisions = Page.objects.filter(content_type=content_type).values_list(
+        "id", flat=True
+    )
+
```
Comment (arslanashraf7) [inline courses/models.py:675]: How about we directly get the certificate pages list from `cms.CertificatePage` without the intervention of content type?

**Review by arslanashraf7 — COMMENTED** [review]
The functionality works well but I added an inline comment for your opinion.

Secondly, If we think of a case where a user might associate a wrong certificate revision with a course run that might make this certificate object invalid. Maybe to be on the safe side we can clean the data in the model before saving the object and check if the certificate revision is actually for the selected course run or not and if the user is selecting an invalid certificate revision?

**Comment by jkachel** [comment]
https://github.com/mitodl/mitxonline/pull/1284/commits/4e81cdf0bac2a6662eb09fc093ec9ac84bd08dc5 adds in a `clean` method to the models that should prevent successful validation if you try to assign a certificate page that's not for the record's courseware object to the record.

File: courses/models.py [hunk 2]
```diff
@@ -758,7 +803,11 @@ class ProgramCertificate(TimestampedModel, BaseCertificate):
 
     program = models.ForeignKey(Program, null=False, on_delete=models.CASCADE)
     certificate_page_revision = models.ForeignKey(
-        PageRevision, null=True, on_delete=models.CASCADE
+        PageRevision,
+        null=True,
+        blank=True,
```
Comment (arslanashraf7) [inline courses/models.py:808]: I think these complaints are about a missing migration. I generated the migration locally and it works.

File: courses/models.py [hunk 3]
```diff
@@ -658,6 +658,21 @@ def save(
...
 
+def limit_to_certificate_pages():
+    """
+    A callable for the limit_choices_to param in the FKs for certificate pages
+    to limit the choices to certificate pages, rather than every page in the
+    CMS.
+    """
+    from cms.models import CertificatePage
+
+    avaialble_revisions = CertificatePage.objects.filter(live=True).values_list(
```
Comment (arslanashraf7) [inline courses/models.py:669]:
```suggestion
    available_revisions = CertificatePage.objects.filter(live=True).values_list(
```

**Review by arslanashraf7 — APPROVED** [review]
Looks good and works great 👍  just a couple of comments (Majorly a migration), feel free to merge afterward.

**Comment by jkachel** [comment]
Last round of changes required some other minor changes to get the tests to complete.

**Review by arslanashraf7 — APPROVED** [review]
👍

**Pull request merged**